### ISSUE TITLE
Drive Manager fails to switch / partition to read-only complaining to existing swap file while the swap is disabled

### ISSUE THREAD

**Issue by dummylabs** [MichaIng/DietPi#2127]
#### Required Information:
- DietPi version	6.16
- Distro version	9,14
- Kernel version	Linux DietPi 4.14.73+ #1148 Mon Oct 1 16:41:23 BST 2018 armv6l GNU/Linux
- SBC device		RPi Zero W (armv6l)
- Power supply used	5V 2.5A Odroid Power Supply

#### Additional Information (if applicable):
- Can this issue be replicated on a fresh installation of DietPi: Yes

#### Steps to reproduce:
1. Disable swap by issuing the command: `/DietPi/dietpi/func/dietpi-set_dphys-swapfile 0`
2. Reboot to be on the safe side
2. Try to make / partition read-only by using drive manager

#### Expected behaviour:
/ partition becomes read-only.

#### Actual behaviour:
Drive manager reports the error: 
```
Failed: The DietPi swapfile is currently located on this drive: - /var/swap                                              
The requested option for this drive is not currently possible.
Please move the swapfile elsewhere, before trying again. 
```
#### Extra details:
1. It seems that swap file does not exist in the mentioned location.
```
rm /var/swap
rm: cannot remove '/var/swap': No such file or directory
```
2. Disabling swap file providing 0 option for corresponding partition in Drive Manager UI makes no difference. It reports swap file state as:
` Swapfile        : [X] | 0MB used on this drive, select to change size`

3. `swapoff -a` doesn't work either

4. It is possible to make /boot partition read-only without issues
5. The command `/DietPi/dietpi/func/dietpi-set_dphys-swapfile 0` gives the same output even being run mutiple times in a row:
```
$ /DietPi/dietpi/func/dietpi-set_dphys-swapfile 0
DietPi-Swapfile
 Mode: Apply
[  OK  ] DietPi-Swapfile | Disable swapfile
[  OK  ] DietPi-Swapfile | swapoff -a
[ INFO ] DietPi-Swapfile | Deleting existing swapfile (/var/swap)
[ INFO ] DietPi-Swapfile | Setting /tmp tmpfs size: 240M
[  OK  ] DietPi-Swapfile | mount -o remount,size=240M tmpfs /tmp
```

**Comment by MichaIng** [comment]
@dummylabs 
Thanks for your report.

Jep indeed this is a bug our side. I also recognized that in drive_manager swap file flag (the `[x]`) is on, even that swap size is zero. It is just checked the (last stored/chosen) swap location, not the size.

As a quick fix, you can manuall edit the swap file location to at best a non-existing location, /dev/null or something, by editing `AUTO_SETUP_SWAPFILE_LOCATION=` in `/DietPi/dietpi.txt`.

We will fix this with v6.17
________
I opened PR about this: https://github.com/Fourdee/DietPi/pull/2129
- Needs testing/checking, if dummy swap location breaks the script somewhere else. But blocking of last stored swap location should be resolved.

**Comment by Fourdee** [comment]
@MichaIng 

Resolved with: https://github.com/Fourdee/DietPi/commit/45d74cbcab438ebc60fa136db9be2234c018a6ca & https://github.com/Fourdee/DietPi/commit/071a84f7dd54e9b6ccf294ffe66e6b098eef83e9

Passed testing.
